### PR TITLE
Add: comprehensive unit test suite for C++ runtime and Python utilities

### PR DESCRIPTION
**Pull request by chenshengxin2026**
## Summary

- Reorganize `tests/ut/cpp/` into domain-based subdirectories: `hierarchical/`, `pto2_a2a3/`, `pto2_a5/`, `types/`, `hardware/`, `platform/`
- Expand C++ GoogleTest coverage from 10 to 30+ test binaries covering PTO2 runtime internals: ring buffer (incl. edge cases), scheduler (state + edge), tensormap (incl. edge), shared memory, ready queue, runtime graph/lifecycle/status, handshake, coupling, orchestrator fatal/submit, boundary edge, dispatch payload, core/submit types, worker manager, and platform stubs
- Add shared `test_helpers.h` header and `stubs/` directory for reusable test infrastructure
- Add 5 new Python UT modules: `test_elf_parser.py`, `test_env_manager.py`, `test_kernel_compiler.py`, `test_runtime_compiler.py`, `test_toolchain_setup.py`
- Update `conftest.py` with improved path setup for isolated UT execution
- Update `docs/testing.md`, `docs/ci.md` for new UT layout; add `docs/ut-test-suite.md` describing the full test inventory
- Minor fixes in `runtime_compiler.py` for testability

### C++ test directory layout

```
tests/ut/cpp/
├── hardware/          # HCCL comm tests (requires_hardware label)
��── hierarchical/      # Orchestrator, ring, scheduler, scope, tensormap, worker_manager
├── platform/          # Platform host_log, memory_allocator stubs
├── pto2_a2a3/         # PTO2 A2A3 runtime: ring_buffer, scheduler, tensormap,
│                      #   shared_memory, ready_queue, coupling, handshake, types, edge cases
├── pto2_a5/           # PTO2 A5 runtime fatal tests
├── stubs/             # Shared test stubs (test_stubs.cpp)
├── types/             # Child memory, PTO types, tensor
├── test_helpers.h     # Shared test helpers
└── CMakeLists.txt     # All test targets with no_hardware / requires_hardware labels
```

### Known defect-exposing tests

Several tests intentionally verify known src defects (documented in the test suite doc). These are preserved as regression anchors per project policy — they will pass once the underlying issues are fixed:
- TensorMap `check_overlap()` dimension mismatch (HIGH)
- TensorMap cleanup slot collision on large windows (HIGH)
- TensorMap lookup saturation with 16+ producers (MEDIUM)
- SharedMemory missing `task_window_size >= 1` validation (LOW)

## Test plan
- [x] C++ UT build succeeds
- [x] C++ UT `no_hardware` tests run
- [x] Python UT passes (44 tests)
- [ ] CI `ut-cpp` job passes on `ubuntu-latest`
- [ ] Simulation tests unaffected (no runtime/platform code changes)

### CODE REVIEW

File: tests/ut/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+endif()
+
+# Stub sources
+set(STUB_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/stubs/test_stubs.cpp)
+
+# Helper: add a test target (only if source file exists)
+function(add_gtest_target name)
+    cmake_parse_arguments(ARG "" "" "SOURCES;EXTRA_SOURCES" ${ARGN})
+    # Check all source files exist
+    foreach(src ${ARG_SOURCES})
```
[the remark this answers is not in the document]
Comment (chenshengxin2026) [reply tests/ut/cpp/CMakeLists.txt]: Fixed — `foreach` now iterates over both `${ARG_SOURCES}` and `${ARG_EXTRA_SOURCES}` to catch missing/misspelled files in either list.
